protocol/contracts/v2/templeLineOfCredit/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,962 @@
+     */
+    function getDebtTokenCache() external view returns (DebtTokenCache memory) {
+        return debtTokenCacheRO();
+    }
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                          INTERNALS                         */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @dev An internal struct used to track the latest storage data (and new updates)
+     * for a given debt token.
+     * This is setup once from storage, and then reads/writes are cheap.
+     */
+    struct DebtTokenCache {
+        /**
+         * @notice This debt token's configuration.
+         */
+        DebtTokenConfig config;
+
+        /**
+         * @notice The total amount that has already been borrowed by all accounts.
+         * This increases as interest accrues or new borrows. 
+         * Decreases on repays or liquidations.
+         */
+        uint128 totalDebt;
+
+        /**
+         * @notice The interest rate as of the last borrow/repay/liquidation.
+         * This last rate is used to accrue interest from that last action time
+         * until the current block.
+         */
+        uint96 interestRate;
+
+        /**
+         * @notice Internal tracking of the accumulated interest as an index starting from 1.0e27
+         * When this accumulator is compunded by the interest rate, the total debt can be calculated as
+         * `updatedTotalDebt = prevTotalDebt * latestInterestAccumulator / prevInterestAccumulator
+         */
+        uint128 interestAccumulator;
+
+        /**
+         * @dev The price of this token as of this block. 
+         * No external price oracles - only dependant on TPI
+         */
+        uint256 price;
+        
+        /**
+         * @notice The maximum amount that the TLC Strategy is allowed to borrow from the TRV
+         * @dev This puts a ceiling on the total amount allowed to be borrowed
+         * by users from TLC. This is used as the denominator in the Utilisation Ratio
+         * (the interest rate calcs are dependant on the UR)
+         */
+        uint256 trvDebtCeiling;
+    }
+    
+    /**
+     * @dev Initialize the DebtTokenCache from storage to this block, for a given token.
+     */
+    function initDebtTokenCache(DebtTokenCache memory _cache) private view returns (bool dirty) {
+
+        // Copies from storage (once)
+        _cache.config = debtTokenConfig;
+
+        // No need to use `encodeUInt128()` here - straight from storage of the same dimension
+        _cache.interestAccumulator = uint128(debtTokenData.interestAccumulator);
+        _cache.totalDebt = debtTokenData.totalDebt;
+        _cache.interestRate = debtTokenData.interestRate;
+
+        // Set the debt ceiling and price.
+        {
+            ITreasuryReservesVault _trv = treasuryReservesVault;
+            _cache.trvDebtCeiling = _trv.strategyDebtCeiling(address(tlcStrategy));
```

Comment:
If `trvDebtCeiling` was changed, at the end of the function, maybe it's needed to call `updateInterestRate()` and set `dirty` to `true`.

Reply:
The approach that I've taken is to expect that when the debt ceiling is increased in TRV, in order to update the interest rates, either:
1. We just wait for a new user borrow/repay/liquidation - the rate is updated at the end of those fns anyway; or
2. If we really need to 'force', the `tlc.updateInterestRates()` function can be explicitly called (can be from the same thing which updates the ceiling)

I have an existing test for (2):
https://github.com/TempleDAO/temple/blob/tlc/protocol/test/forge/v2/templeLineOfCredit/TempleLineOfCreditV2.t.sol#L473

I don't think we need to do track automatically within `initDebtTokenCache()`, because the interest rate is only used within the `initDebtTokenCache()` anyway -- and then the interest rates will be updated at the end of any borrow/repay/liquidation regardless.

Further it would make that function more complicated, and also add more storage - we'd need to need to track the debt ceiling in TLC's debt token storage too.

Let me know what you think.

Reply:
I think both of them are fine and prefer the first option.